{"uid":"a3b92ebd53b67f18","name":"查询用户权限-缺少Authorization","fullName":"testcase.Business Scenario.test_user_permissions.TestUserManager#test_user_permissions","historyId":"207c4599c7799df01061bd211c3aa688","time":{"start":1749900116175,"stop":1749900116325,"duration":150},"status":"broken","statusMessage":"requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)","statusTrace":"self = <Response [500]>, kwargs = {}\n\n    def json(self, **kwargs):\n        r\"\"\"Returns the json-encoded content of a response, if any.\n    \n        :param \\*\\*kwargs: Optional arguments that ``json.loads`` takes.\n        :raises requests.exceptions.JSONDecodeError: If the response body does not\n            contain valid json.\n        \"\"\"\n    \n        if not self.encoding and self.content and len(self.content) > 3:\n            # No encoding set. JSON RFC 4627 section 3 states we should expect\n            # UTF-8, -16 or -32. Detect which one to use; If the detection or\n            # decoding fails, fall back to `self.text` (using charset_normalizer to make\n            # a best guess).\n            encoding = guess_json_utf(self.content)\n            if encoding is not None:\n                try:\n                    return complexjson.loads(self.content.decode(encoding), **kwargs)\n                except UnicodeDecodeError:\n                    # Wrong UTF codec detected; usually because it's not UTF-8\n                    # but some other 8-bit codec.  This is an RFC violation,\n                    # and the server didn't bother to tell us what codec *was*\n                    # used.\n                    pass\n                except JSONDecodeError as e:\n                    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)\n    \n        try:\n>           return complexjson.loads(self.text, **kwargs)\n\n..\\..\\Pyproject\\testProject\\venv\\Lib\\site-packages\\requests\\models.py:974: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\nC:\\Users\\seeki\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\json\\__init__.py:346: in loads\n    return _default_decoder.decode(s)\nC:\\Users\\seeki\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\json\\decoder.py:337: in decode\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <json.decoder.JSONDecoder object at 0x000001ABD4CF7E90>\ns = '<!doctype html><html lang=\"en\"><head><title>HTTP Status 500 – Internal Server Error</title><style type=\"text/css\">bod...e of the root cause is available in the server logs.</p><hr class=\"line\" /><h3>Apache Tomcat/8.5.63</h3></body></html>'\nidx = 0\n\n    def raw_decode(self, s, idx=0):\n        \"\"\"Decode a JSON document from ``s`` (a ``str`` beginning with\n        a JSON document) and return a 2-tuple of the Python\n        representation and the index in ``s`` where the document ended.\n    \n        This can be used to decode a JSON document from a string that may\n        have extraneous data at the end.\n    \n        \"\"\"\n        try:\n            obj, end = self.scan_once(s, idx)\n        except StopIteration as err:\n>           raise JSONDecodeError(\"Expecting value\", s, err.value) from None\nE           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n\nC:\\Users\\seeki\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\json\\decoder.py:355: JSONDecodeError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <test_user_permissions.TestUserManager object at 0x000001AB8B1CCF50>\ncase_info = {'baseInfo': {'api_name': '查询用户权限-缺少Authorization', 'description': '测试缺少Authorization头', 'header': {'Content-Type': 'application/json', 'Cookie': '${get_extract_data(cookie_1)}'}, 'method': 'get', ...}, 'testCase': [{'json': {}}]}\n\n    @allure.story(next(c_id) + \"查询用户权限\")\n    @pytest.mark.run(order=1)\n    @pytest.mark.parametrize('case_info', get_testcase_yaml(\"./testcase/Business Scenario/user_permissions_data.yaml\"))\n    def test_user_permissions(self, case_info):\n        allure.dynamic.title(case_info['baseInfo']['api_name'])\n>       RequestBase().specification_yaml(case_info)\n\ntestcase\\Business Scenario\\test_user_permissions.py:17: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\nbase\\apiutil_business.py:150: in specification_yaml\n    raise e\nbase\\apiutil_business.py:124: in specification_yaml\n    allure.attach(self.allure_attach_response(res.json()), '接口响应信息', allure.attachment_type.TEXT)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <Response [500]>, kwargs = {}\n\n    def json(self, **kwargs):\n        r\"\"\"Returns the json-encoded content of a response, if any.\n    \n        :param \\*\\*kwargs: Optional arguments that ``json.loads`` takes.\n        :raises requests.exceptions.JSONDecodeError: If the response body does not\n            contain valid json.\n        \"\"\"\n    \n        if not self.encoding and self.content and len(self.content) > 3:\n            # No encoding set. JSON RFC 4627 section 3 states we should expect\n            # UTF-8, -16 or -32. Detect which one to use; If the detection or\n            # decoding fails, fall back to `self.text` (using charset_normalizer to make\n            # a best guess).\n            encoding = guess_json_utf(self.content)\n            if encoding is not None:\n                try:\n                    return complexjson.loads(self.content.decode(encoding), **kwargs)\n                except UnicodeDecodeError:\n                    # Wrong UTF codec detected; usually because it's not UTF-8\n                    # but some other 8-bit codec.  This is an RFC violation,\n                    # and the server didn't bother to tell us what codec *was*\n                    # used.\n                    pass\n                except JSONDecodeError as e:\n                    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)\n    \n        try:\n            return complexjson.loads(self.text, **kwargs)\n        except JSONDecodeError as e:\n            # Catch JSON-related errors and raise as requests.JSONDecodeError\n            # This aliases json.JSONDecodeError and simplejson.JSONDecodeError\n>           raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)\nE           requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n\n..\\..\\Pyproject\\testProject\\venv\\Lib\\site-packages\\requests\\models.py:978: JSONDecodeError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"setup_fixture","time":{"start":1749900112194,"stop":1749900112194,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"attachmentStep":false},{"name":"clear_extract","time":{"start":1749900112194,"stop":1749900112196,"duration":2},"status":"passed","steps":[],"attachments":[],"parameters":[],"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"attachmentStep":false},{"name":"datadb_init","time":{"start":1749900112196,"stop":1749900112196,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"attachmentStep":false},{"name":"system_login","time":{"start":1749900112196,"stop":1749900112355,"duration":159},"status":"passed","steps":[],"attachments":[{"uid":"8625a2a589baf6c8","name":"接口名称：用户登录","source":"8625a2a589baf6c8.txt","type":"text/plain","size":12},{"uid":"236b7c6c50818af8","name":"接口地址：http://47.113.145.175:8080/ssm_web/user/login","source":"236b7c6c50818af8.txt","type":"text/plain","size":12},{"uid":"5f119d283fa34634","name":"请求方法：POST","source":"5f119d283fa34634.txt","type":"text/plain","size":12},{"uid":"8a6a49b77f1fbd30","name":"测试用例名称：登录成功","source":"8a6a49b77f1fbd30.txt","type":"text/plain","size":12},{"uid":"bde6903c51f2514","name":"请求参数","source":"bde6903c51f2514.txt","type":"text/plain","size":58},{"uid":"eb82d6670cf5b90f","name":"接口响应信息","source":"eb82d6670cf5b90f.txt","type":"text/plain","size":675}],"parameters":[],"hasContent":true,"shouldDisplayMessage":false,"attachmentsCount":6,"stepsCount":0,"attachmentStep":false},{"name":"start_test_and_end","time":{"start":1749900116175,"stop":1749900116175,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"attachmentStep":false}],"testStage":{"status":"broken","statusMessage":"requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)","statusTrace":"self = <Response [500]>, kwargs = {}\n\n    def json(self, **kwargs):\n        r\"\"\"Returns the json-encoded content of a response, if any.\n    \n        :param \\*\\*kwargs: Optional arguments that ``json.loads`` takes.\n        :raises requests.exceptions.JSONDecodeError: If the response body does not\n            contain valid json.\n        \"\"\"\n    \n        if not self.encoding and self.content and len(self.content) > 3:\n            # No encoding set. JSON RFC 4627 section 3 states we should expect\n            # UTF-8, -16 or -32. Detect which one to use; If the detection or\n            # decoding fails, fall back to `self.text` (using charset_normalizer to make\n            # a best guess).\n            encoding = guess_json_utf(self.content)\n            if encoding is not None:\n                try:\n                    return complexjson.loads(self.content.decode(encoding), **kwargs)\n                except UnicodeDecodeError:\n                    # Wrong UTF codec detected; usually because it's not UTF-8\n                    # but some other 8-bit codec.  This is an RFC violation,\n                    # and the server didn't bother to tell us what codec *was*\n                    # used.\n                    pass\n                except JSONDecodeError as e:\n                    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)\n    \n        try:\n>           return complexjson.loads(self.text, **kwargs)\n\n..\\..\\Pyproject\\testProject\\venv\\Lib\\site-packages\\requests\\models.py:974: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\nC:\\Users\\seeki\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\json\\__init__.py:346: in loads\n    return _default_decoder.decode(s)\nC:\\Users\\seeki\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\json\\decoder.py:337: in decode\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <json.decoder.JSONDecoder object at 0x000001ABD4CF7E90>\ns = '<!doctype html><html lang=\"en\"><head><title>HTTP Status 500 – Internal Server Error</title><style type=\"text/css\">bod...e of the root cause is available in the server logs.</p><hr class=\"line\" /><h3>Apache Tomcat/8.5.63</h3></body></html>'\nidx = 0\n\n    def raw_decode(self, s, idx=0):\n        \"\"\"Decode a JSON document from ``s`` (a ``str`` beginning with\n        a JSON document) and return a 2-tuple of the Python\n        representation and the index in ``s`` where the document ended.\n    \n        This can be used to decode a JSON document from a string that may\n        have extraneous data at the end.\n    \n        \"\"\"\n        try:\n            obj, end = self.scan_once(s, idx)\n        except StopIteration as err:\n>           raise JSONDecodeError(\"Expecting value\", s, err.value) from None\nE           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n\nC:\\Users\\seeki\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\json\\decoder.py:355: JSONDecodeError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <test_user_permissions.TestUserManager object at 0x000001AB8B1CCF50>\ncase_info = {'baseInfo': {'api_name': '查询用户权限-缺少Authorization', 'description': '测试缺少Authorization头', 'header': {'Content-Type': 'application/json', 'Cookie': '${get_extract_data(cookie_1)}'}, 'method': 'get', ...}, 'testCase': [{'json': {}}]}\n\n    @allure.story(next(c_id) + \"查询用户权限\")\n    @pytest.mark.run(order=1)\n    @pytest.mark.parametrize('case_info', get_testcase_yaml(\"./testcase/Business Scenario/user_permissions_data.yaml\"))\n    def test_user_permissions(self, case_info):\n        allure.dynamic.title(case_info['baseInfo']['api_name'])\n>       RequestBase().specification_yaml(case_info)\n\ntestcase\\Business Scenario\\test_user_permissions.py:17: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\nbase\\apiutil_business.py:150: in specification_yaml\n    raise e\nbase\\apiutil_business.py:124: in specification_yaml\n    allure.attach(self.allure_attach_response(res.json()), '接口响应信息', allure.attachment_type.TEXT)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <Response [500]>, kwargs = {}\n\n    def json(self, **kwargs):\n        r\"\"\"Returns the json-encoded content of a response, if any.\n    \n        :param \\*\\*kwargs: Optional arguments that ``json.loads`` takes.\n        :raises requests.exceptions.JSONDecodeError: If the response body does not\n            contain valid json.\n        \"\"\"\n    \n        if not self.encoding and self.content and len(self.content) > 3:\n            # No encoding set. JSON RFC 4627 section 3 states we should expect\n            # UTF-8, -16 or -32. Detect which one to use; If the detection or\n            # decoding fails, fall back to `self.text` (using charset_normalizer to make\n            # a best guess).\n            encoding = guess_json_utf(self.content)\n            if encoding is not None:\n                try:\n                    return complexjson.loads(self.content.decode(encoding), **kwargs)\n                except UnicodeDecodeError:\n                    # Wrong UTF codec detected; usually because it's not UTF-8\n                    # but some other 8-bit codec.  This is an RFC violation,\n                    # and the server didn't bother to tell us what codec *was*\n                    # used.\n                    pass\n                except JSONDecodeError as e:\n                    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)\n    \n        try:\n            return complexjson.loads(self.text, **kwargs)\n        except JSONDecodeError as e:\n            # Catch JSON-related errors and raise as requests.JSONDecodeError\n            # This aliases json.JSONDecodeError and simplejson.JSONDecodeError\n>           raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)\nE           requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n\n..\\..\\Pyproject\\testProject\\venv\\Lib\\site-packages\\requests\\models.py:978: JSONDecodeError","steps":[],"attachments":[{"uid":"ca0d7181102df368","name":"接口地址：http://47.113.145.175:8080/ssm_web/user/getUserPermissions","source":"ca0d7181102df368.attach","type":"text/plain","size":58},{"uid":"69bd29fe23d18a86","name":"接口名：查询用户权限-缺少Authorization","source":"69bd29fe23d18a86.attach","type":"text/plain","size":38},{"uid":"7dc3c5fa71efefd1","name":"请求方法：get","source":"7dc3c5fa71efefd1.attach","type":"text/plain","size":3},{"uid":"4495f87ea3687d4e","name":"请求头：{'Content-Type': 'application/json', 'Cookie': 'JSESSIONID=56F50693F12A336F19543B628C94514C'}","source":"4495f87ea3687d4e.txt","type":"text/plain","size":38},{"uid":"a3e5e240a527a143","name":"请求头信息","source":"a3e5e240a527a143.txt","type":"text/plain","size":93},{"uid":"f2ad39ce664d1f18","name":"测试用例名称：5.测试缺少Authorization头","source":"f2ad39ce664d1f18.txt","type":"text/plain","size":30},{"uid":"e7f4cb3e49a7cdab","name":"请求参数","source":"e7f4cb3e49a7cdab.txt","type":"text/plain","size":12},{"uid":"3db03ba16a7d566d","name":"接口响应信息","source":"3db03ba16a7d566d.txt","type":"text/plain","size":4057},{"uid":"288ba07246f3fd96","name":"log","source":"288ba07246f3fd96.txt","type":"text/plain","size":4992}],"parameters":[],"hasContent":true,"shouldDisplayMessage":true,"attachmentsCount":9,"stepsCount":0,"attachmentStep":false},"afterStages":[{"name":"start_test_and_end::0","time":{"start":1749900116354,"stop":1749900116354,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"attachmentStep":false},{"name":"setup_fixture::0","time":{"start":1749900134145,"stop":1749900134145,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"attachmentStep":false}],"labels":[{"name":"feature","value":"M02_用户管理"},{"name":"story","value":"C02_查询用户权限"},{"name":"tag","value":"run(order=1)"},{"name":"parentSuite","value":"testcase.Business Scenario"},{"name":"suite","value":"test_user_permissions"},{"name":"subSuite","value":"TestUserManager"},{"name":"host","value":"SQG"},{"name":"thread","value":"46620-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"testcase.Business Scenario.test_user_permissions"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"case_info","value":"{'baseInfo': {'description': '测试缺少Authorization头', 'api_name': '查询用户权限-缺少Authorization', 'url': '/ssm_web/user/getUserPermissions', 'method': 'get', 'header': {'Content-Type': 'application/json', 'Cookie': '${get_extract_data(cookie_1)}'}}, 'testCase': [{'case_name': '5.测试缺少Authorization头', 'json': {}, 'validation': [{'contains': {'state': 400}}, {'contains': {'success': False}}, {'contains': {'message': '获取失败'}}]}]}"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[]}],"tags":["run(order=1)"]},"source":"a3b92ebd53b67f18.json","parameterValues":["{'baseInfo': {'description': '测试缺少Authorization头', 'api_name': '查询用户权限-缺少Authorization', 'url': '/ssm_web/user/getUserPermissions', 'method': 'get', 'header': {'Content-Type': 'application/json', 'Cookie': '${get_extract_data(cookie_1)}'}}, 'testCase': [{'case_name': '5.测试缺少Authorization头', 'json': {}, 'validation': [{'contains': {'state': 400}}, {'contains': {'success': False}}, {'contains': {'message': '获取失败'}}]}]}"]}